### PR TITLE
Improve intellisense comment for watchchar event

### PR DESCRIPTION
**Pull request by networkfusion**
## Motivation and Context
* The comment was not correct.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: System.IO.Ports/SerialData.cs [hunk 1]
```diff
@@ -11,12 +11,12 @@ namespace System.IO.Ports
     public enum SerialData
     {
         /// <summary>
-        /// A character was received and placed in the input buffer.
+        /// A character was received in the input buffer.
```
Comment (josesimoes) [inline System.IO.Ports/SerialData.cs:14]: This was correct as it was. The behaviour is that a char was received by the UART and placed in the input buffer. If you want to be picky maybe you can replace with input _stream_ .

File: README.md [hunk 2]
```diff
@@ -140,7 +140,7 @@ private void DataReceivedNormalEvent(object sender, SerialDataReceivedEventArgs
 {
     if (e.EventType == SerialData.WatchChar)
     {
-        // We have our special character at the end of the transmission
+        // We have our special character during the transmission and read was cut short
```
Comment (josesimoes) [inline README.md:143]: Read is never interrupted when the watch char is found. It "just" fires the interrupt to signal that it's present in the input buffer, but the reading goes on.

File: System.IO.Ports/SerialData.cs (hunk 1 above)
Comment (networkfusion) [reply System.IO.Ports/SerialData.cs]: Hopefully made more clear.

File: README.md (hunk 2 above)
Comment (networkfusion) [reply README.md]: Hopefully made more clear.

**Comment by networkfusion** [comment]
Although I still found what I wrote ambiguous...

Is it the case that the `inputstream` has the input buffer up to the watch char, (with the rest of the buffer being available at the next event)
OR
the `inputstream` contains the watchchar somewhere within it?

I guess that is the difference between `watchchar` and `newLine`, so want to make sure the wording is specific...

File: README.md [hunk 3]
```diff
@@ -140,7 +140,8 @@ private void DataReceivedNormalEvent(object sender, SerialDataReceivedEventArgs
 {
     if (e.EventType == SerialData.WatchChar)
     {
-        // We have our special character at the end of the transmission
+        // We have our special character during the transmission and interupt was raised,
```
Comment (josesimoes) [inline README.md:143]:
```suggestion
        // We have our special character during the transmission and an event was raised,
```
Call me picky but in .NET these are events...

File: System.IO.Ports/SerialData.cs [hunk 4]
```diff
@@ -11,12 +11,12 @@ namespace System.IO.Ports
...
     {
         /// <summary>
-        /// A character was received and placed in the input buffer.
+        /// A character was received and placed in the input stream.
         /// </summary>
         Chars = 1,
 
         /// <summary>
-        ///  The end of file character was received and placed in the input buffer.
+        ///  The character to watch was received in the input buffer and prior content placed in the input stream.
```
Comment (josesimoes) [inline System.IO.Ports/SerialData.cs:19]: I'm not English native, but "(...) and prior content placed (...)" sounds confusing to me and I don't understand what it it means exactly. 

Edge case: only the watch char was received, there is no other content in the stream, so... My point being: we don't have to try to explain here every possible situation and workflow that could happen. For this one we have to say exactly and just that: the watch char was detected in the input stream. Period. Everything beyond that just adds confusion.

**Comment by josesimoes** [comment]
> Is it the case that the `inputstream` has the input buffer up to the watch char, (with the rest of the buffer being available at the next event) OR the `inputstream` contains the watchchar somewhere within it?
> 
> I guess that is the difference between `watchchar` and `newLine`, so want to make sure the wording is specific...

This is a but more complex than _just_ that...

Up receiving a wathc char the event is raised AND the UART keeps receiving data (if any). Between the time the event is created and that the event handler in C# is actually running, there could be more data arriving at the UART, that will be placed in the input stream and that will be available by the time the event handler is being executed. 

Morover, the new line is not meant to fire any event, so no concern on explaining or even considering it in the event context.

File: README.md [hunk 5]
```diff
@@ -140,7 +140,8 @@ private void DataReceivedNormalEvent(object sender, SerialDataReceivedEventArgs
 {
     if (e.EventType == SerialData.WatchChar)
     {
-        // We have our special character at the end of the transmission
+        // A specified character was received during the transmission and a notification event was raised.
```
Comment (josesimoes) [inline README.md:143]: was received during the transmission... 🤪 

Why not simply "The char to watch has been received"?
This code is at the event handler, therefore it has been raised. No doubts about it. Why the need to state the obvious?

File: README.md [hunk 6]
```diff
@@ -140,7 +140,8 @@ private void DataReceivedNormalEvent(object sender, SerialDataReceivedEventArgs
...
+        // The rest of the buffer is still being received in the background, so might not be the end of transmission when reading the stream.
```
Comment (josesimoes) [inline README.md:144]: There is only "rest of the buffer" if anything is expected after this... so this can be misleading or confusing!

For example I usually use this to detect new lines at the end of AT replies. There's the payload and the watch char at the end. Done! If anything is being received after that it will be another packet.

File: System.IO.Ports/SerialData.cs [hunk 7]
```diff
@@ -8,16 +8,29 @@ namespace System.IO.Ports
...
+    /// The EventType property contains one of the values from the `SerialData` enumeration.
+    /// </remarks>
     public enum SerialData
     {
         /// <summary>
         /// A character was received and placed in the input buffer.
         /// </summary>
         Chars = 1,
 
+        // /// <summary>
```
Comment (josesimoes) [inline System.IO.Ports/SerialData.cs:23]: Why is this needed here?

Comment (networkfusion) [reply System.IO.Ports/SerialData.cs:23]: We should really support that `End of File`... But the current enum field was taken, and it is a catch 22 between removal and change...

Comment (josesimoes) [reply System.IO.Ports/SerialData.cs:23]: 👀  https://social.msdn.microsoft.com/Forums/vstudio/en-US/6aaf1c70-46c6-485b-a769-2389da7684ee/ioportsserialport-datareceived-eventtypeeof

Comment (networkfusion) [reply System.IO.Ports/SerialData.cs:23]: Interesting... But that does not say we should not support it... Just fix it... But that is not the reason for this PR. If prefered, I can remove it, but feel it currently serves as a valid placeholder (given possibly reserved enum field).

Comment (josesimoes) [reply System.IO.Ports/SerialData.cs:23]: I don't think we are able to support EOF... what's your use case for it?

Comment (networkfusion) [reply System.IO.Ports/SerialData.cs:23]: After reading https://stackoverflow.com/questions/12483711/serialdata-eof-circumstances I agree that it is not something we should not support. I will remove it.

**Review by josesimoes — APPROVED** [review]
